pycloudlib/cloud.py

```diff
@@ -67,8 +74,17 @@ def __init__(
         else:
             self.tag = validate_tag(tag)
 
+    def __enter__(self):
+        """Enter context manager for this class."""
+        return self
+
+    def __exit__(self, _type, _value, _trackback):
+        """Cleanup context manager for this class."""
+        exceptions = self.clean()
+        print_exception_list(exceptions)
```

Comment:
Good question. My initial thinking is I don't think clean failures need to be catastrophic, but I could also see the case where something is broken and you don't want to keep going. I think it'd be more common that a cleanup failure is something minor, like timeout before resource is gone, so I think it's better to continue rather than exit.

You have any strong opinions?

Reply:
I think we definitely should exit(1) here. As BEnjamin's comment on the original issue was:
` Regarding overall verbosity of different implementations, I care less about number of lines, and care more about the chance of accidentally leaking / orphaning cloud resources, especially with default parameters.`

Reply:
Rather than exit(1), I think we raise a separate custom exception instead. Something like `CleanupException`.

exit(1) is a pretty broad hammer that gives nobody the option to recover.

Sound good?

Reply:
Much better.